suplemon/viewer.py

```diff
@@ -50,76 +51,38 @@ def __init__(self, app, window):
         self.x_scroll = 0
         self.cursors = [Cursor()]
 
-        # Lexer for translating tokens to strings
-        self.lexer = None
-        # Built in syntax definition (for commenting etc.)
-        self.syntax = None
-        # Normal Pygments lexer
-        self.pygments_syntax = None
+        # Copy/paste buffer
+        self.buffer = []
+
+        # Runnable methods
+        self.operations = {
+            "arrow_right": self.arrow_right,              # Arrow Right
+            "arrow_left": self.arrow_left,                # Arrow Left
+            "arrow_up": self.arrow_up,                    # Arrow Up
+            "arrow_down": self.arrow_down,                # Arrow Down
+            "jump_left": self.jump_left,                  # Ctrl + Left
+            "jump_right": self.jump_right,                # Ctrl + Right
+            "jump_up": self.jump_up,                      # Ctrl + Up
+            "jump_down": self.jump_down,                  # Ctrl + Down
+            "page_up": self.page_up,                      # Page Up
+            "page_down": self.page_down,                  # Page Down
+            "home": self.home,                            # Home
+            "end": self.end,                              # End
+            "find": self.find_query                       # Ctrl + F
+        }
```

Comment:
Ah, sorry. I missed a parameter for `run_operation` in the example. It would receive an `operation` parameter as it does now in the code:

https://github.com/richrd/suplemon/blob/819fe6e5e630d8fcb21870498ea8ddac90d99f3f/suplemon/viewer.py#L637-L646

```python
  def run_operation(self, operation):
    # Simplified for example
    if operation in self.operations:
      self.operations.arrow_right()
```